### PR TITLE
#6904: Stop propagation of certain events in QuickBar

### PR DESCRIPTION
**Pull request by fregante**
## What does this PR do?

- Fixes https://github.com/pixiebrix/pixiebrix-extension/issues/6904

## Demo

https://github.com/pixiebrix/pixiebrix-extension/assets/1402241/460ff1d2-f954-4c76-be1a-062dbba30726

## Checklist

- [x] Add tests
- [x] New files added to `src/tsconfig.strictNullChecks.json` (if possible)
- [x] Designate a primary reviewer: @grahamlangford

### CODE REVIEW

File: src/components/quickBar/QuickBarApp.tsx [hunk 1]
```diff
@@ -141,10 +141,10 @@ const KBarComponent: React.FC = () => {
...
           >
             <Stylesheets href={faStyleSheet} mountOnLoad>
-              <FocusLock>
+              <StopPropagation onKeyPress onKeyDown onKeyUp>
                 <KBarSearch style={searchStyle} />
                 <QuickBarResults />
-              </FocusLock>
+              </StopPropagation>
             </Stylesheets>
           </EmotionShadowRoot.div>
```
Comment (fregante) [inline src/components/quickBar/QuickBarApp.tsx:149]: This shadow root is open btw, I don’t think it's intentional:

Screenshot

**Comment by twschiller** [comment]
@grahamlangford this change will need to be tested on Salesforce, Pipedrive, Zendesk and all other platforms using hotkeys

@fregante what's the reasoning behind dropping focus lock?

File: src/components/quickBar/QuickBarApp.tsx (hunk 1 above)
Comment (twschiller) [reply src/components/quickBar/QuickBarApp.tsx:149]: Hmm, it was switched from closed to open when we switched libraries: https://github.com/pixiebrix/pixiebrix-extension/pull/5750/files#diff-50f2a63c2398eca75310e434d2dc90a762b23dd6591766e05b68d38a5e00c3a8L122. That wasn't flagged during that PR

I agree it should likely be closed - host sites shouldn't need to snoop on the individual events within the Quick Bar

**Comment by fregante** [comment]
I looked at the original implementation and it mentioned "stealing key presses" rather than focus. The focus change was probably a consequence of the key press going through.

**Comment by fregante** [comment]
The QuickBar seems to work correctly even without it, automatically catching the keypresses and restoring focus. Here I had a `setInterval` that focused the first input on the page, and no keypresses were lost:


https://github.com/pixiebrix/pixiebrix-extension/assets/1402241/8c8d14f2-9aec-4621-b7d6-9b642d25b8d2

The focus lock component might be useful in the future, but it seems that kbar + `StopPropagation` handles this scenario

File: src/components/quickBar/QuickBarApp.tsx (hunk 1 above)
Comment (fregante) [reply src/components/quickBar/QuickBarApp.tsx:150]: I closed it now 👌

**Comment by fregante** [comment]
I created a playground here: https://pbx.vercel.app/site-with-hotkeys/

As mentioned in the issue, we can't really prevent the document from seeing the events if they use `capture`:

Screenshot 13

I'll add an `input` blocker too

**Comment by twschiller** [comment]
>  I looked at the original implementation and it mentioned "stealing key presses" rather than focus. The focus change was probably a consequence of the key press going through.

I'll have to try this on Salesforce, etc. IIRC, the issue is that some sites have key<X> event handlers on document with `capture: true`. So they're able to handle/stop the event before our quick bar

**Pull request merged**